### PR TITLE
Notfoundpage #306

### PR DESCRIPTION
**Pull request by davidholyko**
reference to #306

### Before submitting this PR, please use netlify to make sure:
- [xx] The app looks okay on web
- [ ] The app looks okay on phone
- [x] Filter by category still works

### If you might be concerned about these things, also check:
- [ ] The app looks okay in ipad
- [ ] Search still works

#### Reference to issue
404 Page not rendered on specific pages;
* fix: render 404 page on examples
1. /:notAValidResource
2. /:resource/:notAValidPath

#### What PR does:
1. Uses Switch component to render NotFoundPage when specific path is not found
2. If path is correct and resource is not found

### CODE REVIEW

File: src/App/AppContainer.js [hunk 1]
```diff
@@ -118,6 +126,7 @@ class AppContainer extends Component {
                         <div>
                             <Route exact path='/:resource/admin' render={(props) => <AdminPage currentPosition={this.state.position} />} />
                             <Route exact path='/:resource/' render={(props) => <MapPage currentPosition={this.state.position} />} />
+							<Route path='/:resource/' render={NotFoundPage} />
```
Comment (RajChaudhry) [inline src/App/AppContainer.js]: looks good, just take out this line so the 404 page isn't rendered on every page, and it's perfect.

File: src/App/AppContainer.js [hunk 2]
```diff
@@ -70,15 +73,16 @@ class AppContainer extends Component {
 
         getAllSites.then(sites => {
             resourceSheetId = sheetIdFromPath(sites, resourcePath) || envSheetId;
-        
+
             if(resourceSheetId == null){
-                alert("Error: Unable to find resource '" + resourcePath + "'");
+				// alert("Error: Unable to find resource '" + resourcePath + "'");
+				this.setState({isValidRoute: false})
```
Comment (RajChaudhry) [inline src/App/AppContainer.js]: also just remember to end all the JS parts with semicolons, even though it still works

File: src/App/AppContainer.js [hunk 3]
```diff
@@ -93,16 +91,17 @@ class AppContainer extends Component {
     }
 
     render() {
-		let { isFetchingResource } = this.props;
-
-		if (!this.state.isValidRoute) {
```
Comment (RajChaudhry) [inline src/App/AppContainer.js]: now the 404 doesnt pop up you are on the correct path but wrong route, i think it's because you remove this line of code.

**Comment by RajChaudhry** [comment]
/:resource/admin incorrectly returns a 404 page. for example: /revere/admin will return 404. just add                 <Route path = '/:resource/admin' exact component={AppContainer} dispatch={store.dispatch}/>
to your index.js should fix

**Comment by timtheguy** [comment]
Hi @davidholyko - thanks for your contribution. Could you edit your pull request to include (1) a reference to the issue you are resolving with this pull request and (b) a brief description of what your PR does? This helps us paint a picture of how code got to where it is over the course of the development lifecycle. Thank you in advance!

**Pull request merged**